api/db/base.js

```diff
@@ -0,0 +1,191 @@
+const logger = require('../logger')('db base model');
+const pick = require('lodash.pick');
+
+class ValidationError extends Error {
+  constructor(message) {
+    super(message);
+
+    this.error = { error: message };
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  get statusCode() {
+    return 400;
+  }
+}
```

Comment:
This might be a premature optimization, but I needed a way to signal out that this was a validation error rather than some other exception.  I tried a couple of things, then I thought hey, what if we had a set of custom errors and they could set a statusCode?  I'm not sold on this approach, though...

Reply:
👍